net: pkt: Fix possible division by zero

miyatsu opened the pull request:
When net debugging is enabled, the count variable is initialized to -1.
This may cause division by zero if there is only one fragment in pkt.
Solve this by setting the count to 0 and checking the value before the
print at the end of the function.

Successfully tested on STM32F407 SoC.

Signed-off-by: Ding Tao <miyatsu@qq.com>

jukkar reviewed (commented):
Looks good, we should probably also check that count is not zero at the end of the function as if for some reason there are no fragments, the count is 0 and you have the same issue.

Also the commit message is a bit hard to read, perhaps something like this could be written there:

"When net debugging is enabled, the count variable is initialized to -1. This may cause division by zero if there is only one fragment in pkt. Solve this by setting the count to 0 and checking the value before the print at the end of the function.

Successfully tested on STM32F407 SoC."

jukkar commented on subsys/net/ip/net_pkt.c:
```diff
@@ -253,7 +253,7 @@ void net_pkt_print_frags(struct net_pkt *pkt)
 {
```
Typo in commit msg, s/okt/pkt/

Also add check to last print, it could be something like
count ? (total * 100) / (count * frag_size) : 0

miyatsu replied:
It's done, please review.

miyatsu commented:
Hi @jukkar 
I don't think we need checking the count value before the print at the end of the function, since there is a NET_ASSERT(pkt->frags); at line 265. The code definite gonna run into the while loop, count++ executed.
My english is not good, thanks for the suggest commit message.

miyatsu committed:
net: pkt: Fix possible division by zero

When net debugging is enabled, the count variable is initialized to -1.
This may cause division by zero if there is only one fragment in pkt.
Solve this by setting the count to 0 and checking the value before the
print at the end of the function.

Successfully tested on STM32F407 SoC.

Signed-off-by: Ding Tao <miyatsu@qq.com>

pfalcon reviewed (approved):
Ack for the change, count was set to -1 by me in a745832dc48780092e92778c9a2b7f9b931fe1c1 *before* the net_pkt refactor, and indeed should have gone back to 0 after that. Extra check is also good.
